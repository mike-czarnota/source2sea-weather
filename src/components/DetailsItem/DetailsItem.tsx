interface IDetailsItemProps {
  label: string;
  value: string;
}

const DetailsItem: React.FC<IDetailsItemProps> = ({ label, value }) => (
  <div className="flex flex-col bg-white p-5">
    <span className="italic text-gray-600">{label}</span>
    <span className="text-2xl font-bold text-gray-800">{value}</span>
  </div>
);

export default DetailsItem;
