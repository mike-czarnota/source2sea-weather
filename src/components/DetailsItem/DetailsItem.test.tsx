import pretty from 'pretty';
import { render, screen } from '@testing-library/react';
import DetailsItem from './DetailsItem';

describe('testing DetailsItem.test.tsx', () => {
  it('should render properly with passed props', () => {
    expect(
      pretty(
        render(<DetailsItem label="some label" value="some value" />)
          .baseElement.innerHTML
      )
    ).toMatchSnapshot();

    const label = screen.getByText('some label');
    expect(label).toBeInTheDocument();

    const value = screen.getByText('some value');
    expect(value).toBeInTheDocument();
  });
});
