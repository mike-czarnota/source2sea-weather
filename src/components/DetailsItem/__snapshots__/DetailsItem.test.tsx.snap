// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing DetailsItem.test.tsx should render properly with passed props 1`] = `
"<div>
  <div class=\\"flex flex-col bg-white p-5\\"><span class=\\"italic text-gray-600\\">some label</span><span class=\\"text-2xl font-bold text-gray-800\\">some value</span></div>
</div>"
`;
